source asana  - fix schema related failure

the connector's build [is failing](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/builds.md). This issue tracks fixing it.

It looks like there is a problem with schema correctness for the following streams (issue with `format` for fields):
- sections
- stories
- tasks
- users
```
------------------------------ Captured log call -------------------------------
ERROR    root:test_core.py:158 The sections stream has the following schema errors:
'2021-06-29T09:42:21.036Z' is not of type 'null', 'array'

Failed validating 'type' in schema['properties'***['created_at'***:
    ***'format': 'date-time', 'type': ['null', 'array'***

On instance['created_at'***:
    '2021-06-29T09:42:21.036Z'
ERROR    root:test_core.py:158 The stories stream has the following schema errors:
'2021-06-29T09:45:51.926Z' is not of type 'null', 'array'

Failed validating 'type' in schema['properties'***['created_at'***:
    ***'format': 'date-time', 'type': ['null', 'array'***

On instance['created_at'***:
    '2021-06-29T09:45:51.926Z'
ERROR    root:test_core.py:158 The tasks stream has the following schema errors:
' https://github.com/airbytehq/airbyte/pull/4376' is not of type 'null', 'boolean'

Failed validating 'type' in schema['properties'***['custom_fields'***['items'***['properties'***['text_value'***:
    ***'type': ['null', 'boolean'***

On instance['custom_fields'***[12***['text_value'***:
    ' https://github.com/airbytehq/airbyte/pull/4376'
ERROR    root:test_core.py:158 The users stream has the following schema errors:
[***'gid': '1200272750255812'*** ***'gid': '1199152354181012'*** is not of type 'null', 'object'

Failed validating 'type' in schema['properties'***['workspaces'***:
    ***'properties': ***'gid': ***'type': ['null', 'string'***
                    'name': ***'type': ['null', 'string'***
                    'resource_type': ***'type': ['null', 'string'***
     'type': ['null', 'object'***

On instance['workspaces'***:
    [***'gid': '1200272750255812'*** ***'gid': '1199152354181012'***
```

Also, you need to fix the `invalid_config.json` file so that the keys match the `spec.json`.
```
E           docker.errors.ContainerError: Command 'check --config tap_config.json' in image 'sha256:a882940262858c1389406dc7c32d4da79e443647f4effa6778f3cdb6f16d7226' returned non-zero exit status 1: Traceback (most recent call last):
E             File "/airbyte/integration_code/main.py", line 33, in <module>
E               launch(source, sys.argv[1:***)
E             File "/usr/local/lib/python3.7/site-packages/airbyte_cdk/entrypoint.py", line 125, in launch
E               for message in source_entrypoint.run(parsed_args):
E             File "/usr/local/lib/python3.7/site-packages/airbyte_cdk/entrypoint.py", line 96, in run
E               check_config_against_spec_or_exit(config, source_spec, logger)
E             File "/usr/local/lib/python3.7/site-packages/airbyte_cdk/sources/utils/schema_helpers.py", line 141, in check_config_against_spec_or_exit
E               raise Exception("Config validation error: " + validation_error.message) from None
E           Exception: Config validation error: 'access_token' is a required property
```

--- a/airbyte-config/init/src/main/resources/config/STANDARD_SOURCE_DEFINITION/d0243522-dccf-4978-8ba0-37ed47a0bdbf.json
+++ b/airbyte-config/init/src/main/resources/config/STANDARD_SOURCE_DEFINITION/d0243522-dccf-4978-8ba0-37ed47a0bdbf.json
@@ -2,6 +2,6 @@
   "sourceDefinitionId": "d0243522-dccf-4978-8ba0-37ed47a0bdbf",
   "name": "Asana",
   "dockerRepository": "airbyte/source-asana",
-  "dockerImageTag": "0.1.1",
+  "dockerImageTag": "0.1.2",
   "documentationUrl": "https://docs.airbyte.io/integrations/sources/asana"
 }

--- a/airbyte-config/init/src/main/resources/seed/source_definitions.yaml
+++ b/airbyte-config/init/src/main/resources/seed/source_definitions.yaml
@@ -7,7 +7,7 @@
 - sourceDefinitionId: d0243522-dccf-4978-8ba0-37ed47a0bdbf
   name: Asana
   dockerRepository: airbyte/source-asana
-  dockerImageTag: 0.1.1
+  dockerImageTag: 0.1.2
   documentationUrl: https://docs.airbyte.io/integrations/sources/asana
   sourceType: api
 - sourceDefinitionId: 686473f1-76d9-4994-9cc7-9b13da46147c

--- a/airbyte-integrations/connectors/source-asana/CHANGELOG.md
+++ b/airbyte-integrations/connectors/source-asana/CHANGELOG.md
@@ -1,4 +1,0 @@
-# Changelog
-
-## 0.1.0
-Initial Release.

--- a/airbyte-integrations/connectors/source-asana/Dockerfile
+++ b/airbyte-integrations/connectors/source-asana/Dockerfile
@@ -12,5 +12,5 @@ RUN pip install .
 ENV AIRBYTE_ENTRYPOINT "python /airbyte/integration_code/main.py"
 ENTRYPOINT ["python", "/airbyte/integration_code/main.py"]
 
-LABEL io.airbyte.version=0.1.1
+LABEL io.airbyte.version=0.1.2
 LABEL io.airbyte.name=airbyte/source-asana

--- a/airbyte-integrations/connectors/source-asana/integration_tests/invalid_config.json
+++ b/airbyte-integrations/connectors/source-asana/integration_tests/invalid_config.json
@@ -1,3 +1,3 @@
 {
-  "todo-wrong-field": "this should be an incomplete config file, used in standard tests"
+  "access_token": "<wrong_access_token>"
 }

--- a/airbyte-integrations/connectors/source-asana/source_asana/schemas/sections.json
+++ b/airbyte-integrations/connectors/source-asana/source_asana/schemas/sections.json
@@ -11,7 +11,7 @@
       "type": ["null", "string"]
     },
     "created_at": {
-      "type": ["null", "array"],
+      "type": ["null", "string"],
       "format": "date-time"
     },
     "project": {

--- a/airbyte-integrations/connectors/source-asana/source_asana/schemas/stories.json
+++ b/airbyte-integrations/connectors/source-asana/source_asana/schemas/stories.json
@@ -8,7 +8,7 @@
       "type": ["null", "string"]
     },
     "created_at": {
-      "type": ["null", "array"],
+      "type": ["null", "string"],
       "format": "date-time"
     },
     "created_by": {

--- a/airbyte-integrations/connectors/source-asana/source_asana/schemas/tasks.json
+++ b/airbyte-integrations/connectors/source-asana/source_asana/schemas/tasks.json
@@ -328,7 +328,7 @@
             "type": ["null", "string"]
           },
           "text_value": {
-            "type": ["null", "boolean"]
+            "type": ["null", "string"]
           },
           "type": {
             "type": ["null", "string"]

--- a/airbyte-integrations/connectors/source-asana/source_asana/schemas/users.json
+++ b/airbyte-integrations/connectors/source-asana/source_asana/schemas/users.json
@@ -34,16 +34,19 @@
       }
     },
     "workspaces": {
-      "type": ["null", "object"],
-      "properties": {
-        "gid": {
-          "type": ["null", "string"]
-        },
-        "resource_type": {
-          "type": ["null", "string"]
-        },
-        "name": {
-          "type": ["null", "string"]
+      "type": ["null", "array"],
+      "items": {
+        "type": ["null", "object"],
+        "properties": {
+          "gid": {
+            "type": ["null", "string"]
+          },
+          "resource_type": {
+            "type": ["null", "string"]
+          },
+          "name": {
+            "type": ["null", "string"]
+          }
         }
       }
     }

--- a/docs/integrations/sources/asana.md
+++ b/docs/integrations/sources/asana.md
@@ -61,6 +61,6 @@ to obtain Personal Access Token for your account.
 
 | Version | Date       | Pull Request | Subject |
 | :------ | :--------  | :-----       | :------ |
-| 0.1.2   | 2021-09-24 | [6402](https://github.com/airbytehq/airbyte/pull/6402) | Fix SAT tests |
+| 0.1.2   | 2021-09-24 | [6402](https://github.com/airbytehq/airbyte/pull/6402) | Fix SAT tests: update schemas and invalid_config.json file |
 | 0.1.1   | 2021-06-09 | [3973](https://github.com/airbytehq/airbyte/pull/3973) | Add entrypoint and bump version for connector |
 | 0.1.0   | 2021-05-25 | [3510](https://github.com/airbytehq/airbyte/pull/3510) | New Source: Asana |

--- a/docs/integrations/sources/asana.md
+++ b/docs/integrations/sources/asana.md
@@ -56,3 +56,11 @@ The Asana connector should not run into Asana API limitations under normal usage
 Please follow these [steps](https://developers.asana.com/docs/personal-access-token)
 to obtain Personal Access Token for your account.
 
+
+## Changelog
+
+| Version | Date       | Pull Request | Subject |
+| :------ | :--------  | :-----       | :------ |
+| 0.1.2   | 2021-09-24 | [6402](https://github.com/airbytehq/airbyte/pull/6402) | Fix SAT tests |
+| 0.1.1   | 2021-06-09 | [3973](https://github.com/airbytehq/airbyte/pull/3973) | Add entrypoint and bump version for connector |
+| 0.1.0   | 2021-05-25 | [3510](https://github.com/airbytehq/airbyte/pull/3510) | New Source: Asana |

